### PR TITLE
Coinbase buy widget integration

### PR DESCRIPTION
**Pull request by hiddentao**
Coinbase buy widget integration.

### CODE REVIEW

**Comment by hiddentao** [comment]
Preliminary widget integration code done. But widget doesn't show because we need for Coinbase to approve our local app URLs...

In order to get the Coinbase buy widget showing within Mist we need for our local dev URLs as well packaged Mist/Wallet app URLs to be added as `frame-src` items as part of the Content Security Policy `frame-ancestors` header -> https://developer.mozilla.org/en/docs/Web/Security/CSP/CSP_policy_directives#frame-ancestors

**Comment by hiddentao** [comment]
I have contacted Coinbase via the Slack channel asking about this.

**Comment by hiddentao** [comment]
Need a US debit card to do the full test. I don't have one :/

File: app/client/lib/thirdparty/coinbase-widget.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    self.modalIframeStyle = function() {
+        return "\n      transition: all 0.3s ease-out;\n      background-color: transparent;\n      border: 0px none transparent;\n      display: none;\n      position: fixed;\n      visibility: visible;\n      margin: 0px;\n      padding: 0px;\n      left: 0px;\n      top: 0px;\n      width: 100%;\n      height: 100%;\n      z-index: 9999;\n    ";
+    };
+
+    self.modalIframe = function() {
+        var e = self.generateParams();
+        return "<iframe src='" + self.domain + "/" + e + "'\n                    id='coinbase_modal_iframe'\n                    name='coinbase_modal_iframe'\n                    style='" + self.modalIframeStyle() + "'\n                    scrolling='no'\n                    allowtransparency='true' frameborder='0'>\n      </iframe>";
+    };
+
+    // self.buttonIframeStyle = function() {
```
Comment (alexvandesande) [inline app/client/lib/thirdparty/coinbase-widget.js]: can we remove these comments?

**Comment by alexvandesande** [comment]
I haven't been able to test it either, but from a design perspective I made the changes I wanted

**Comment by alexvandesande** [comment]
Please ping when you're able to test it

**Comment by alexvandesande** [comment]
Hey Ram, what's the status now? From the testing it feels everything is working properly!

**Comment by hiddentao** [comment]
@alexvandesande This is ready.

**Comment by user061783** [comment]
Link: <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=2>; rel="next",
  <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=34>; rel="last"

**Pull request merged**